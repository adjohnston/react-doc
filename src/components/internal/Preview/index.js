import React from 'react'
import PropTypes from 'prop-types'

const Preview = ({ Component }) => {
  return (
    <section>
      <Component />
    </section>
  )
}

Preview.propTypes = {
  Component: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.element,
  ]).isRequired,
}

export { Preview as InternalPreview }
