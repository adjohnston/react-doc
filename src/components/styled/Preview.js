import React, { PropTypes } from 'react'
import styled from 'styled-components'

const Preview = styled.section`
  display: flex;
  justify-content: center;
  margin: 0 0 1rem;
  padding: 5rem;
  background: #eee;
`

Preview.propTypes = {
  children: PropTypes.element.isRequired
}

export default Preview
