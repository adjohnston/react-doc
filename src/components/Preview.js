import React from 'react'

import { Custom } from 'src/context'
import { InternalPreview } from 'src/components/internal/Preview'

const Preview = ({ children }) => {
  return (
    <Custom>
      {({ Component }) =>
        children ? (
          children({ Component })
        ) : (
          <InternalPreview Component={Component} />
        )
      }
    </Custom>
  )
}

export { Preview }
