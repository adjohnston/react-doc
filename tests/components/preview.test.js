import React from 'react'
import renderer from 'react-test-renderer'
import Preview from 'components/Preview'

const Test = () => <div>Just a test</div>

test('it renders correctly', () => {
  const props = {
    swatches: [],
    children: <Test />
  }

  const tree = renderer
    .create(<Preview {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it renders a list of swatch buttons if swatches are passed', () => {
  const props = {
    swatches: ['white', 'black', '#f35'],
    children: <Test />
  }

  const tree = renderer
    .create(<Preview {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it renders a different background colour when a swatch is clicked', () => {
  const props = {
    swatches: ['white', 'black', '#f35'],
    children: <Test />
  }

  const component = renderer
    .create(<Preview {...props} />)

  component
    .getInstance()
    .setSwatch('#f35')

  expect(component.toJSON()).toMatchSnapshot()
})
