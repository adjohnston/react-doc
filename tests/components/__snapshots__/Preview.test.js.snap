// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="stncl__preview"
  style={null}
>
  <div>
    Just a test
  </div>
</div>
`;

exports[`it should render correctly when given a template 1`] = `
<div
  className="some-other-class"
  style={
    Object {
      "background": "A test component",
    }
  }
>
  <div>
    Just a test
  </div>
</div>
`;
